tests/unit/test_tiler.py

```diff
@@ -409,16 +424,29 @@ def _random_tile_coordinates(self, request):
 
 
 class Describe_GridTiler:
-    @pytest.mark.parametrize("level", (2, -2))
-    def it_constructs_from_args(self, level, request):
+    @pytest.mark.parametrize("level, mpp", ((2, None), (-2, None), (2, 0.5)))
+    def it_constructs_from_args(self, level, mpp, request):
         _init = initializer_mock(request, GridTiler)
 
-        grid_tiler = GridTiler((512, 512), level, True, 80, 0, "", ".png")
+        grid_tiler = GridTiler((512, 512), level, True, 80, 0, "", ".png", mpp=mpp)
 
-        _init.assert_called_once_with(ANY, (512, 512), level, True, 80, 0, "", ".png")
+        _init.assert_called_once_with(
+            ANY, (512, 512), level, True, 80, 0, "", ".png", mpp=mpp
+        )
         assert isinstance(grid_tiler, GridTiler)
         assert isinstance(grid_tiler, Tiler)
 
+    @pytest.mark.parametrize(
+        "level, mpp, expected_level, expected_mpp",
+        ((2, None, 2, None), (None, 0.5, 0, 0.5), (2, 0.5, 0, 0.5)),
+    )
+    def mpp_supercedes_level(self, level, mpp, expected_level, expected_mpp):
```

Comment:
with this name this test will not be collected

Reply:
i changed it ... it wasn't me :D

Reply:
actually we had 2 i changed only 1 XD

Reply:
lol